src/python/pants/backend/shell/shell_command.py

```diff
@@ -140,7 +140,7 @@ async def prepare_shell_command_process(
                 binary_name=tool,
                 search_path=search_path,
             )
-            for tool in {*tools, *["mkdir", "ln"]}
+            for tool in (*tools, *["mkdir", "ln"])
```

Comment:
My guess is that the intent here was to dedup, so consider using `OrderedSet`?

Reply:
Indeed it was 👍🏽